packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt

```diff
@@ -0,0 +1,25 @@
+package io.flutter.ios_platform_images
+
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+class IosPlatformImagesPlugin: MethodCallHandler {
+  companion object {
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "plugins.flutter.io/ios_platform_images")
+      channel.setMethodCallHandler(IosPlatformImagesPlugin())
+    }
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    if (call.method == "getPlatformVersion") {
+      result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    } else {
+      result.error("AndroidNotSupported", "This plugin is for iOS only.", null)
```

Comment:
Are you sure you want to make this plugin iOS-specific (as opposed to leaving the door open to supporting Android in the future)? It seems like the API is simple enough at this point to support a lowest common denominator Android reimplementation, although I think it makes sense to focus on iOS at the outset.

Possibly relevant wisdom (though it doesn't apply as much to packages that are outside the Flutter SDK)
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-exposing-api-cliffs
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-exposing-api-oceans

Reply:
I argued this point and came around to the other side.  The conventional wisdom doesn't stand in the add2app scenario, it was formed around full Flutter apps.

The point of the plugin is to support sharing assets in an existing iOS app with a subset of the UI that is written in Flutter.  That means if the iOS API supplies 3 different ways of loading images, it isn't sufficient to support 2 of them and ask users to edit their iOS app in order to integrate Flutter.

This plugin is pointless in a non-add2app scenario.  When users want cross-platform image loading they should move their assets into Flutterland.

Reply:
It sounds like you've made up your mind on this. I think if I were maintaining this I would lean towards either a very focused plugin (covering only iOS, only one direction) or a plugin that covers both iOS and Android and tries to cover as many image sharing use cases as possible. Right now you're sort of halfway and it isn't clear what functionality should be provided.

But I don't feel strongly about it and if you want to land it like this it's fine by me. I'd like @amirh to provide a second review though.